src/lfortran/semantics/ast_body_visitor.cpp

```diff
@@ -273,6 +273,12 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 if (!ASRUtils::is_character(*a_form_type)) {
                         throw SemanticError("`form` must be of type, Character or CharacterPointer", x.base.base.loc);
                 }
+            } else {
+                const std::unordered_set<std::string> unsupported_args {"iostat", "iomsg", "err", "blank", "access", \
+                                                                            "recl", "fileopt", "action", "position"};
+                if (unsupported_args.find(m_arg_str) == unsupported_args.end()) {
+                    throw SemanticError("invalid argument: `" + m_arg_str + "` supplied", x.base.base.loc);
+                }
```

Comment:
Why not just:
```suggestion
                throw SemanticError("invalid argument: `" + m_arg_str + "` supplied", x.base.base.loc);
```

Reply:
If I do that then unsupported kwargs like `action, blank, recl` (i.e. the ones mentioned in `unsupported_args`) would raise a Semantic Error, which they shouldn't, right?

This made question for why `FileOpen` doesn't have `action, blank, recl` etc. in it's definition in ASR.asdl? defined here: https://github.com/lfortran/lfortran/blob/main/src/libasr/ASR.asdl#L48

Reply:
I see. Then do something like:
```suggestion
                const std::unordered_set<std::string> unsupported_args {"iostat", "iomsg", "err", "blank", "access", \
                                                                            "recl", "fileopt", "action", "position"};
                if (unsupported_args.find(m_arg_str) == unsupported_args.end()) {
                    throw SemanticError("Invalid argument: `" + m_arg_str + "` supplied", x.base.base.loc);
                } else {
                    throw SemanticError("Argument: `" + m_arg_str + "` not implemented yet", x.base.base.loc);
                }
```
And add tests for both cases.

Reply:
Do you think we should just raise a warning instead of an error? I mean, for someone who already uses these kwargs, their code would break and also would need changes in our test-suite (many instance of use of "access" and "action".